dashboard/app/controllers/lessons_controller.rb

```diff
@@ -2,6 +2,20 @@ class LessonsController < ApplicationController
   load_and_authorize_resource
 
   before_action :require_levelbuilder_mode_or_test_env, except: [:show]
+  before_action :disallow_legacy_script_levels, only: [:edit, :update]
+
+  # Script levels which are not in activity sections will not show up on the
+  # lesson edit page, in which case saving the edit page would cause those
+  # script levels to be lost. Prevent this by disallowing editing in this case.
+  # This helps avoid losing data from existing scripts by accidentally editing
+  # them with the new lessons editor on levelbuilder.
+  def disallow_legacy_script_levels
+    return if [:development, :adhoc].include? rack_env
+    return unless @lesson.script_levels.reject(&:activity_section).any?
+    raise CanCan::AccessDenied.new(
+      "cannot edit lesson #{@lesson.id} because it contains legacy script levels"
+    )
+  end
```

Comment:
warning: if you go in and edit an existing lesson that had script levels in it in development, and you save the lesson edit page, it will wipe out all of those script levels. Of course, since you are in development, you can get them back easily by doing `rake seed:single_script`. 

I wanted to keep this protection in for development too, but then you get into a bootstrapping problem where you aren't allowed to create an empty lesson so the only way to create anything to start with is from the command line.

Reply:
The empty lesson problem is now fixed right?

Reply:
Good point -- that means i could add this protection back for development and adhoc. Will do!